### PR TITLE
if an event is an error, pass the exception object

### PR DESCRIPTION
**Pull request by cxreg**
Upstream PR https://github.com/JustinTulloss/zeromq.node/pull/578 has not been responded to at this time.  I propose to merge this to our fork for now.

### CODE REVIEW

**Comment by pmuellr** [comment]
Still a little bothered by having a separate upstream PR, esp since we don't know the state of it.

I believe the last changes to storage to produce better error messages for already bound zmq ports will be a good enough test to make sure nothings broken between node/npm-zmq and nsolid - make sure you run the storage integration tests w/nsolid to make sure nothings b0rken ...

Other than that, LGTM, but not really familiar with the NAN usage in the natives ...

**Comment by cxreg** [comment]
The NAN usage was copied from neighboring code, so it should be ok

I'll ping upstream one more time...

File: lib/index.js [hunk 1]
```diff
@@ -531,7 +531,15 @@ Socket.prototype.monitor = function(interval, numOfEvents) {
...
 
     self._zmq.onMonitorEvent = function(event_id, event_value, event_endpoint_addr) {
-      self.emit(events[event_id], event_value, event_endpoint_addr);
+      var type = events[event_id];
+      var ex;
+      switch (type) {
+        case "bind_error":
+        case "accept_error":
+        case "close_error":
+          ex = self._zmq.getException();
```
Comment (thlorenz) [inline lib/index.js:540]: probably wanna put a `break` here to prevent fall through if more cases are introduced

File: binding.cc [hunk 2]
```diff
@@ -1425,6 +1427,10 @@ namespace zmq {
     return;
   }
 
+  NAN_METHOD(Socket::GetException) {
+    info.GetReturnValue().Set(ExceptionFromError());
```
Comment (thlorenz) [inline binding.cc]: `ExceptionFromError` is some magic global function I suppose working on some global error object?
Gotta love that C style programming ;)

**Comment by thlorenz** [comment]
In general LGTM

**Comment by cxreg** [comment]
Upstream accepted a modified version of this but it's not merged yet.  Keeping this open for a few more days to see how that shakes out and when it lands

**Comment by cxreg** [comment]
Upstream fix merged and landed, closing this

**Pull request closed without merging**